java/test/src/main/java/test/Ice/metrics/AllTests.java

```diff
@@ -527,13 +524,25 @@ static MetricsPrx allTests(test.TestHelper helper, CommunicatorObserverI obsv)
       map = toMap(serverMetrics.getMetricsView("View").returnValue.get("Connection"));
       test(map.get("holding").current == 1);
 
-      metrics.ice_getConnection().close(com.zeroc.Ice.ConnectionClose.GracefullyWithWait);
+      // `close` blocks the calling thread until the connection closure is complete.
```

Comment:
Why is this different that when we use GracefullyWithWait?

Reply:
GracefullyWithWait did not mean wait until the connection is closed. It meant wait until all two-way invocations have completed (the wait part) then start a graceful connection closure. But don't wait for its completion.

And plain Gracefully meant start a graceful connection closure without worrying about outstanding two-way invocations (that in practice would fail as a result of this "graceful" closure).

Reply:
Yes, like @bernardnormier says.
The old `close(GracefullyWithWait)` would only wait for invocations to complete, _not_ for the connection to be closed.
The new `close()` waits for the connection to be closed (either because all invocations completed, or a timeout was hit).